ZAPD/GameConfig.cpp

```diff
@@ -0,0 +1,138 @@
+#include "GameConfig.h"
+
+#include <functional>
+#include "Utils/Directory.h"
+#include "Utils/File.h"
+#include "Utils/Path.h"
+#include "tinyxml2.h"
+
+void GameConfig::ReadTexturePool(const std::string& texturePoolXmlPath)
+{
+	tinyxml2::XMLDocument doc;
+	tinyxml2::XMLError eResult = doc.LoadFile(texturePoolXmlPath.c_str());
+
+	if (eResult != tinyxml2::XML_SUCCESS)
+	{
+		fprintf(stderr, "Warning: Unable to read texture pool XML with error code %i\n", eResult);
+		return;
+	}
+
+	tinyxml2::XMLNode* root = doc.FirstChild();
+
+	if (root == nullptr)
+		return;
+
+	for (tinyxml2::XMLElement* child = root->FirstChildElement(); child != nullptr;
+	     child = child->NextSiblingElement())
+	{
+		if (std::string(child->Name()) == "Texture")
+		{
+			std::string crcStr = std::string(child->Attribute("CRC"));
+			fs::path texPath = std::string(child->Attribute("Path"));
+			std::string texName = "";
+
+			uint32_t crc = strtoul(crcStr.c_str(), nullptr, 16);
+
+			texturePool[crc].path = texPath;
+		}
+	}
+}
+
+void GameConfig::GenSymbolMap(const std::string& symbolMapPath)
+{
+	auto symbolLines = File::ReadAllLines(symbolMapPath);
+
+	for (std::string symbolLine : symbolLines)
+	{
+		auto split = StringHelper::Split(symbolLine, " ");
+		uint32_t addr = strtoul(split[0].c_str(), nullptr, 16);
+		std::string symbolName = split[1];
+
+		symbolMap[addr] = symbolName;
```

Comment:
```suggestion
		symbolMap[addr] = std::move(symbolName);
```
would allow the symbolName to be moved into the map without any extra allocation or copy.